build.py

```diff
@@ -34,36 +62,43 @@ def print_usage():
     configs_string = "\n  ".join(configs)
-    cmake_name = os.path.basename(CMAKE)
-    generator_name = CMAKE_GENERATOR.lower()
-    default_config_name = DEFAULT_CONFIG_NAME
-    debug_config_name = DEBUG_CONFIG_NAME
-    print(f"""Usage: {script_name} [BUILD [BUILD ...]] [--all] [--debug] [MAKE_OPTIONS]
-
-Build one or more predefined build configurations of Fast Downward. Each build
-uses {cmake_name} to compile the code using {generator_name} . Build configurations
-differ in the parameters they pass to {cmake_name}. By default, the build uses all
-available cores if this number can be determined. Use the "-j" option for
-{cmake_name} to override this default behaviour.
-
-Build configurations
+    epilog = f"""build configurations:
   {configs_string}
 
---all         Alias to build all build configurations.
---debug       Alias to build the default debug build configuration.
---help        Print this message and exit.
-
-Make options
-  All other parameters are forwarded to the build step.
-
-Example usage:
-  ./{script_name}                     # build {default_config_name} in #cores threads
-  ./{script_name} -j4                 # build {default_config_name} in 4 threads
-  ./{script_name} debug               # build debug
-  ./{script_name} --debug             # build {debug_config_name}
-  ./{script_name} release debug       # build release and debug configs
-  ./{script_name} --all VERBOSE=true  # build all build configs with detailed logs
-""")
+example usage:
+  {script_name}                     # build {DEFAULT_CONFIG_NAME} with #cores parallel processes
+  {script_name} -j4                 # build {DEFAULT_CONFIG_NAME} with 4 parallel processes
+  {script_name} debug               # build debug
+  {script_name} --debug             # build {DEBUG_CONFIG_NAME}
+  {script_name} release debug       # build release and debug configs
+  {script_name} --all VERBOSE=true  # build all build configs with detailed logs
+"""
+
+    parser = argparse.ArgumentParser(
+        description=description, epilog=epilog, formatter_class=RawHelpFormatter)
+    parser.add_argument("--debug", action="store_true",
+                        help="alias to build the default debug build configuration")
+    parser.add_argument("--all", action="store_true",
+                        help="alias to build all build configurations")
+    remaining_args = parser.add_argument("arguments", nargs=argparse.REMAINDER,
+                        help="build configurations (see below) or build options")
+    remaining_args.completer = complete_arguments
+
+    if HAS_ARGCOMPLETE:
+        argcomplete.autocomplete(parser)
+    args, unparsed_args = parser.parse_known_args()
+
+    if hasattr(args, "remaining_args"):
+        args.remaining_args += unparsed_args
+    else:
+        args.remaining_args = unparsed_args
+    return args
+
+
+def complete_arguments(prefix, parsed_args, **kwargs):
+    split_args(parsed_args)
```

Comment:
This looks like it modifies parsed_args in place. Should `complete_arguments` do this as a side effect? If yes, should we document this somewhere?

Reply:
It is fine to do this because the process will stop after generating suggestions for the tab completion. I'll add a comment.